editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx

```diff
@@ -531,13 +533,21 @@ export const AnalyticsProvider: React.FC<{ children: React.ReactNode }> = ({
    * Extract allowlist answers from breadcrumb data
    * e.g. data set automatically by components such as DrawBoundary
    */
-  function getData(breadcrumb: Store.userData) {
+  function getData(
+    breadcrumb: Store.userData,
+  ): Partial<Record<AllowListKey, any>> | undefined {
     const dataSetByNode = breadcrumb.data;
-    if (!dataSetByNode) return [];
+    if (!dataSetByNode) return;
+
+    const filteredEntries = Object.entries(dataSetByNode)
+      .filter(
+        ([key, value]) =>
+          ALLOW_LIST.includes(key as AllowListKey) && Boolean(value),
```

Comment:
See previous comment on using a type guard (`is` function) - we can probably use one here to avoid casting twice.

Reply:
Should also be sorted by: 9bccce1bbfd6b0271de9d7796f7b3bed4791a779